### PR TITLE
#160451517 Users should be able to like a specific comment

### PR DESCRIPTION
**Pull request by samdiano**
#### What does this PR do?
This pull request creates functionality to like and dislike a specific comment and replies

#### Description of Task to be completed?
- Write tests for liking a specific comment
- Write the code to like a specific comment

#### How should this be manually tested?
To like a comment: /api/v1/articles/:articleSlug/comments/:commentId/like -> POST request
To like a comment reply: /api/v1/articles/:articleSlug/comments/:commentId/reply/:replyId/like -> POST request
To dislike a comment: /api/v1/articles/:articleSlug/comments/:commentId/dislike -> POST request
To dislike a comment reply: /api/v1/articles/:articleSlug/comments/:commentId/reply/:replyId/dislike -> POST request

#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
#160451517

### CODE REVIEW

File: server/controllers/articleController.js [hunk 1]
```diff
@@ -69,7 +69,14 @@ export default class ArticleController {
         ],
         include: [
           { model: Tag, as: 'Tags', through: 'ArticlesTags' },
-          { model: Like, as: 'likes' }
+          {
```
Comment (samdiano) [inline server/controllers/articleController.js:72]: I included the details of the user that liked an article

File: server/controllers/likeController.js [hunk 2]
```diff
@@ -14,12 +14,43 @@ export default class LikeController {
   * @returns {object} - status, message and reaction details
   */
   static likeArticle(req, res) {
-    // Check the params passed by user to determine what function to be performed
-    const like = req.params.likeType === 'like';
-    const dislike = req.params.likeType === 'dislike';
-    // message to be sent to user depending on function performed
-    const message = like || dislike ? `you ${req.params.likeType}d the article`
-      : 'you unliked the article';
+    const message = 'you liked this article';
```
Comment (samdiano) [inline server/controllers/likeController.js:17]: refactored the implementation of liking an article as the former implementation causes conflicts

File: server/helpers/article.js [hunk 3]
```diff
@@ -1,4 +1,4 @@
-
+/* eslint-disable no-restricted-globals */
```
Comment (samdiano) [inline server/helpers/article.js]: added this because eslint flagged isNaN as a restricted global method

File: server/models/articlecomment.js [hunk 4]
```diff
@@ -20,6 +20,11 @@ export default (sequelize, DataTypes) => {
       foreignKey: 'commentId',
       onDelete: 'CASCADE',
     });
+    ArticleComment.hasMany(models.LikeComment, {
```
Comment (samdiano) [inline server/models/articlecomment.js:23]: Added relationship between ArticleComment and likeComment models

File: server/routes/api/article.js [hunk 5]
```diff
@@ -18,6 +18,7 @@ router.get('/articles/:slug', ArticleController.getSingleArticle);
 router.delete('/articles/:slug', checkToken, ArticleController.deleteArticle);
 router.put('/articles/:slug', checkToken, upload.single('imgUrl'), ArticleValidations.validateUpdateArticle, ArticleController.updateArticle);
 router.post('/articles/:slug/rate', checkToken, RateValidations.validateRating, RateController.rateArticle);
-router.post('/articles/:slug/:likeType', checkToken, LikeController.likeArticle);
+router.post('/articles/:slug/like', checkToken, LikeController.likeArticle);
```
Comment (samdiano) [inline server/routes/api/article.js:22]: Changed routes to define like and dislike explicitly

File: server/tests/article.test.js [hunk 6]
```diff
@@ -170,28 +170,6 @@ describe('Articles', () => {
         done();
       });
   });
```
Comment (samdiano) [inline server/tests/article.test.js:172]: removed like tests from article.test.js and moved it to like.test.js

File: server/tests/comment.test.js [hunk 7]
```diff
@@ -70,6 +70,17 @@ describe('Comment', () => {
       });
   });
 
+  it('Should get all comments of an article', (done) => {
```
Comment (samdiano) [inline server/tests/comment.test.js:73]: Added test to get all comments. It was not tested before

**Review by slimsolz — COMMENTED** [review]
LGTM

**Comment by kampkelly** [comment]
LGTM

**Review by s-akinrele — CHANGES_REQUESTED** [review]
As discussed

**Review by onwuvic — COMMENTED** [review]
LGTM

File: server/controllers/likeCommentController.js [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+    }).then((comment) => {
+      if (comment) {
+        LikeComment.findOrCreate({
+          where: {
+            userId: req.user.id,
+            commentId: req.params.commentId
+          },
+          defaults: {
+            like: true,
+            dislike: false
```
Comment (s-akinrele) [inline server/controllers/likeCommentController.js]: please remove this line of code

File: server/controllers/likeReplyController.js [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+    }).then((reply) => {
+      if (reply) {
+        LikeReply.findOrCreate({
+          where: {
+            userId: req.user.id,
+            replyId: req.params.replyId
+          },
+          defaults: {
+            like: false,
+            dislike: true
```
Comment (s-akinrele) [inline server/controllers/likeReplyController.js]: remove this also

**Review by s-akinrele — APPROVED** [review]

**Pull request merged**